spec/app/domain/authentication/authenticate_spec.rb

```diff
@@ -135,64 +124,123 @@ def authenticator(pass:)
+            validate_user_can_access_webservice: mock_validate_user_can_access_webservice(validation_succeeded: true),
+            validate_webservice_is_whitelisted:  mock_validate_webservice_is_whitelisted(validation_succeeded: false),
+            validate_origin:                     mocked_origin_validator,
+            token_factory:                       mocked_token_factory,
+            audit_log:                           mocked_audit_logger
+          ).call(
+            authenticator_input:    input_,
+            authenticators:         authenticators,
+            enabled_authenticators: two_authenticator_env
+          )
+        end
 
         it "raises an error" do
-          allow(mocked_security_validator).to receive(:call)
-                                                .and_raise('FAKE_SECURITY_ERROR')
-
-          expect { subject }.to raise_error(
-                                  /FAKE_SECURITY_ERROR/
-                                )
+          expect { subject }.to(
+            raise_error(
+              validate_webservice_is_whitelisted_error
+            )
+          )
         end
       end
 
-      context "when origin validation fails" do
-        let(:audit_success) { false }
-
-        it "raises an error" do
-          allow(mocked_origin_validator).to receive(:call)
-                                              .and_raise('FAKE_ORIGIN_ERROR')
+      context "when webservice validations succeed" do
+        let(:audit_success) { true }
+
+        subject do
+          input_ = Authentication::AuthenticatorInput.new(
+            authenticator_name: 'authn-always-pass',
+            service_id:         nil,
+            account:            'my-acct',
+            username:           'my-user',
+            credentials:        'my-pw',
+            origin:             '127.0.0.1',
+            request:            nil
+          )
+
+          Authentication::Authenticate.new(
+            validate_user_can_access_webservice: mock_validate_user_can_access_webservice(validation_succeeded: true),
+            validate_webservice_is_whitelisted:  mock_validate_webservice_is_whitelisted(validation_succeeded: true),
+            validate_origin:                     mocked_origin_validator,
+            token_factory:                       mocked_token_factory,
+            audit_log:                           mocked_audit_logger
+          ).call(
+            authenticator_input:    input_,
+            authenticators:         authenticators,
+            enabled_authenticators: two_authenticator_env
+          )
+        end
 
-          expect { subject }.to raise_error(
-                                  /FAKE_ORIGIN_ERROR/
-                                )
+        it "returns a new token" do
+          expect(subject).to(
+            equal(
+              a_new_token
+            )
+          )
         end
+
+        it_behaves_like "raises an error when origin validation fails"
```

Comment:
Why is this here? The context is `when webservice validations succeed` so how does that relate to an request coming from a blocked origin?

Reply:
this is inside the vanilla flow, just like before. We verify that it fails when origin validation fails (and everything else succeeds)